### DISCUSSION TITLE
How to: Route params to component props ?

### DISCUSSION THREAD

**Discussion by awacode21** [nuxt/nuxt#25175] [Questions]
Hi guys,

Vue router provides the possibility for a route to declare "props: true" so that route params get provided as component props to the rendered component.

I wonder how to do that with nuxt 3. Was not able to find anything in documentation or google search. Is this feature not possible with nuxt 3??

Thanks in advance, best regards, Annick

**Comment by patrickcate** [comment]
I have the same question. Vue Router docs for this functionality is at: https://router.vuejs.org/guide/essentials/passing-props

I was not able to find any mention of this in the Nuxt docs.

**Comment by tachibana-shin** [comment]
We can do this by creating a file in `app/router.config.ts`

```typescript
import type { RouterConfig } from "nuxt/schema"

export default {
  routes(_routes) {
    // biome-ignore lint/suspicious/noAssignInExpressions: <explanation>
    // biome-ignore lint/complexity/noForEach: <explanation>
    _routes.forEach((route) => void (route.props = true))
    return _routes
  }
} satisfies RouterConfig
```

**Comment by xjccc** [comment]
https://github.com/nuxt/nuxt/pull/29586
Look at this. Will update in nuxt@3.14
